Add snowplow-cli data products (#1092)
snowplow/documentation: docs/pipeline-components-and-applications/cli/index.md

---
title: Snowplow CLI
date: 2024-11-25
sidebar_label: Snowplow CLI
sidebar_position: 999
---
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

> `snowplow-cli` means to bring Snowplow Console into the command line. It allows you to download your data structures and data products to yaml/json files and publish them back to console. This enables the git-ops workflow, with reviews and brancing.

Reviewer: ```suggestion
`snowplow-cli` brings data management elements of Snowplow Console into the command line. It allows you to download your data structures and data products to yaml/json files and publish them back to console. This enables git-ops-like workflows, with reviews and brancing.
```

# Install

> Snowplow CLI can be installed with [homebrew](https://brew.sh/)

Reviewer: ```suggestion
Snowplow CLI can be installed with [homebrew](https://brew.sh/):
```

```
brew install snowplow-product/taps/snowplow-cli
```

Verify the installation with
```
snowplow-cli --help
```

For systems where homebrew is not available binaries for multiple platforms can be found in [releases](https://github.com/snowplow-product/snowplow-cli/releases).

Example installation for `linux_x86_64` using `curl`

```bash
curl -L -o snowplow-cli https://github.com/snowplow-product/snowplow-cli/releases/latest/download/snowplow-cli_linux_x86_64
chmod u+x snowplow-cli
```

Verify the installation with
```
./snowplow-cli --help
```

# Configure

You will need three values.

> API Key Id and API Key Secret are generated from the [credentials section](https://console.snowplowanalytics.com/credentials) in BDP Console.

Reviewer: ```suggestion
An API Key Id and the corresponding API Key (secret), which are generated from the [credentials section](https://console.snowplowanalytics.com/credentials) in BDP Console.
```

> Organization Id can be retrieved from the URL immediately following the .com when visiting BDP console:

Reviewer: ```suggestion
The organization ID, which can be retrieved from the URL immediately following the .com when visiting BDP console:
```

  ```bash
  SNOWPLOW_CONSOLE_API_KEY_ID=********-****-****-****-************
  SNOWPLOW_CONSOLE_API_KEY=********-****-****-****-************
  SNOWPLOW_CONSOLE_ORG_ID=********-****-****-****-************
  ```

  </TabItem>
  <TabItem value="defaultconfig" label="$HOME/.config/snowplow/snowplow.yml" >

  ```yaml
  console:
    api-key-id: ********-****-****-****-************
    api-key: ********-****-****-****-************
    org-id: ********-****-****-****-************
  ```

  </TabItem>
  <TabItem value="args" label="inline arguments" >

  ```bash
  ./snowplow-cli data-structures --api-key-id ********-****-****-****-************ --api-key ********-****-****-****-************ --org-id ********-****-****-****-************
  ```

  </TabItem>
</Tabs>

Snowplow CLI defaults to yaml format. It can be changed to json by either providing a `--output-format json` flag or setting the `output-format: json` config value. It will work for all commands where it matters, not only for `generate`.

# Use cases

- [Manage your data structures with snowplow-cli](/docs/understanding-tracking-design/managing-your-data-structures/cli/index.md)

> - [Set up a github CI/CD pipeline to manage data structures](/docs/recipes/recipe-data-structures-in-git/index.md)

Reviewer: ```suggestion
- [Set up a github CI/CD pipeline to manage data structures and data products](/docs/recipes/recipe-data-structures-in-git/index.md)
```
